content/fast/adaptive-serving-based-on-network-quality/codelab-adaptive-serving-based-on-network-quality.md

```diff
@@ -0,0 +1,172 @@
+---
+page_type: glitch
+title: Adaptive serving based on network quality
+author: mihajlija
+description: |
+  In this codelab you’ll learn how to use the Network Information API to adapt
+  your content based on the network quality.
+web_updated_on: 2019-01-31
+web_published_on: 2019-01-31
+glitch: adaptive-serving-netinfo-starter
+---
+
+# Adaptive serving based on network quality
+
+In this codelab you’ll learn how to adapt your content based on the network quality. This page’s background video should load only when users are on a fast network. On slower networks an image should load instead.
```

Comment:
Suggestion to remove 'shoulds' and you are missing a comma. Something like: "This page's background video loads only when users are on a fast network. On slower networks, an image loads instead."

Reply:
Replaced 'should' with 'will' because video to image adaptation isn't implemented at the beginning of the codelab. At this point, _this page's background video loads regardless of the network quality_. It _will load only when users are on a fast network_, but only after people finish the codelab. :)